File: frontend/src/api/lessonRecord.ts
/**
 * 课时消耗记录 API - 老王写的，别瞎改
 */
import request from './index'
import type { ListResponse } from '@/types'

export interface LessonRecord {
  id: number
  enrollment_id: number
  schedule_id: number | null
  record_date: string
  hours: number
  type: string
  notes: string | null
  created_time: string
  student_name: string | null
  class_plan_name: string | null
  teacher_name: string | null
  schedule_date: string | null
}

export interface LessonRecordListResponse {
  items: LessonRecord[]
  total: number
  page: number
  page_size: number
}

/**
 * 根据报名ID获取课时消耗记录
 */
export function getLessonRecordsByEnrollment(
  enrollmentId: number,
  params?: { page?: number; page_size?: number }
) {
  return request.get<ListResponse<LessonRecord>>(
    `/lesson-records/by-enrollment/${enrollmentId}`,
    { params }
  ).then(res => res.data)
}

/**
 * 根据学生ID获取课时消耗记录
 */
export function getLessonRecordsByStudent(
  studentId: number,
  params?: { page?: number; page_size?: number; class_plan_id?: number }
) {
  return request.get<ListResponse<LessonRecord>>(
    `/lesson-records/by-student/${studentId}`,
    { params }
  ).then(res => res.data)
}

File: backend/app/schemas/lesson_record.py
"""
LessonRecord schemas - 课时消耗记录
"""
from datetime import date, datetime
from decimal import Decimal
from typing import Optional, List

from pydantic import BaseModel, Field


class LessonRecordBase(BaseModel):
    """课时消耗记录基础"""
    enrollment_id: int = Field(..., description="报名记录ID")
    schedule_id: Optional[int] = Field(None, description="排课ID")
    record_date: date = Field(..., description="消耗日期")
    hours: Decimal = Field(..., description="消耗课时数")
    type: str = Field(default="schedule", description="类型")
    notes: Optional[str] = Field(None, description="备注")


class LessonRecordCreate(LessonRecordBase):
    """创建课时消耗记录"""
    pass


class LessonRecordResponse(LessonRecordBase):
    """课时消耗记录响应"""
    id: int
    created_time: datetime
    # 关联信息
    student_name: Optional[str] = None
    class_plan_name: Optional[str] = None
    teacher_name: Optional[str] = None
    schedule_date: Optional[date] = None

    model_config = {"from_attributes": True}


class LessonRecordListResponse(BaseModel):
    """课时消耗记录列表响应"""
    items: List[LessonRecordResponse]
    total: int
    page: int
    page_size: int
